aopy/analysis/base.py

```diff
@@ -2372,4 +2376,261 @@ def simulate_ideal_trajectories(targets, origin=[0.0, 0.0, 0.0], resolution=1000
+
+            elec_pos, _, _ = aopy.data.load_chmap()
+            n_elec = len(elec_pos)
+            total_trials = 1600
+            n_trials = 50
+            np.random.seed()
+            data = 0.5*np.ones((total_trials,n_elec))
+            data += np.random.normal(0.5, size=(total_trials,n_elec))
+            data[:,n_elec//4:n_elec//2] += 0.1
+
+            dists = aopy.analysis.calc_statistic_random_trials(data, n_trials=n_trials)
+            self.assertEqual(np.shape(dists), (len(data)//n_trials, n_elec))
+
+            # Test that the distribution means are close to the original data mean
+            mean = np.mean(data, axis=0)
+            dists_mean = np.mean(dists, axis=0)
+            dists_std = np.std(dists, axis=0)
+            for i in range(n_elec):
+                self.assertAlmostEqual(mean[i], dists_mean[i], delta=0.1)
+            clim = (np.min(dists_mean), np.max(dists_mean))
+
+            plt.figure(figsize=(5,2), dpi=300)
+            plt.subplot(1,3,1)
+            im = aopy.visualization.plot_spatial_drive_map(mean, 
+                                            elec_data=True, cmap='Grays')
+            im.set_clim(*clim)
+            plt.axis('off')
+            plt.colorbar(im, shrink=0.5)
+            plt.title('data mean')
+
+            plt.subplot(1,3,2)
+            im = aopy.visualization.plot_spatial_drive_map(dists_mean, 
+                                            elec_data=True, cmap='Grays')
+            im.set_clim(*clim)
+            plt.axis('off')
+            plt.colorbar(im, shrink=0.5)
+            plt.title('dist means')
+
+            plt.subplot(1,3,3)
+            im = aopy.visualization.plot_spatial_drive_map(dists_std, 
+                                            elec_data=True, cmap='Grays')
+            plt.axis('off')
+            plt.colorbar(im, shrink=0.5)
+            plt.title('dist std')
+
+        .. image:: _images/calc_statistic_random_trials.png
+    '''
+    all_trials = np.arange(len(data))
+    np.random.shuffle(all_trials)
+    n_div = len(data)//n_trials
+    if n_div < 1:
+        raise ValueError(f"Not enough trials in data ({len(data)}) to perform the calculation with n_trials ({n_trials}).")
+    trials = np.reshape(all_trials[:n_div*n_trials], (n_div, n_trials))
+
+    dist = []
+    for k in range(n_div):
+        try:
+            dist.append(statistic(data[trials[k]]))
+        except KeyError:
+            dist.append(statistic(data.iloc[trials[k]].reset_index(drop=True)))
+
+    return np.array(dist)
+
+def _calc_statistic_random_trials(args):
+    '''
+    Helper function to unpack arguments for parallel processing.
+    '''
+    return calc_statistic_random_trials(*args)
+
+def calc_trial_bootstraps(data, n_trials=300, n_bootstraps=30, 
+                          statistic=partial(np.mean, axis=0), parallel=False):
+    '''
+    Repeatedly call :func:`~aopy.analysis.calc_statistic_random_trials` to generate
+    statistics over groups of trials.
```

Comment:
I would like more description about what the groups of trials are and how they're chosen.